### DISCUSSION TITLE
can you nest wildcards ?

### DISCUSSION THREAD

**Discussion by simonmcnair** [adieyal/sd-dynamic-prompts#689] [General]
I was just looking at https://github.com/adieyal/sd-dynamic-prompts/blob/main/collections/artists/European%20Art/neoclassical/neoclassical.txt

and tried doing __neoclassical__ and it didn't use it , but did expand it to __artists/European Art/neoclassical/neoclassical/neoclassical__,

cheers

**Comment by DonMischo** [comment]
From my experience the wildcard it refers to are missing. But nesting is possible. ;)

**Reply by Codyport** [reply]
I have had success with nested wildcards, so the functionality is certainly there.  However after more than one level of nesting, or more than a simple degree of complexity, or more than a very limited number of batches (not images in a batch - actual batches) all wildcards are ignored and the Dynamic Prompt extension is effectively disabled.

I speculate that I'm triggering a memory constraint.  The behavior seems to indicate that the prompts for all images in a batch are created at the time the batch is started not as each image is generate - or something similar.  Just speculation, but the fact that the success of the prompt is tied to a combination of the number of batches run and the degree of nesting indicates some relationship between the two.

Start simple, with only one layer of nesting, simple prompts and one image per batch - that works - then add to that and see when you trip a limit.

For reference I'm using A1111 1.8.0 on a Kubuntu install, 4070/12Gb and 64Gb RAM.

**Reply by uorufu** [reply]
On every batch generation, all image main prompts are first generated before starting. 

Adetailer prompts are only generated right before the image is being detailed.

As for running out of memory. I think the prompts are all appended into a list before generation and I am pretty sure python lists can be appended like 500 million times. So I doubt that is the issue for you. There might be something silly breaking large batches.
